core/consensus/babe/src/lib.rs

```diff
@@ -336,18 +382,18 @@ impl<H, B, C, E, I, Error, SO> slots::SimpleSlotWorker<B> for BabeWorker<C, E, I
 			let signature = pair.sign(header_hash.as_ref());
 			let signature_digest_item = <DigestItemFor<B> as CompatibleDigestItem>::babe_seal(signature);
 
-			// When we building our own blocks we always author on top of the
-			// current best according to `SelectChain`, therefore our own block
-			// proposal should always become the new best.
 			BlockImportParams {
 				origin: BlockOrigin::Own,
 				header,
 				justification: None,
 				post_digests: vec![signature_digest_item],
 				body: Some(body),
 				finalized: false,
-				auxiliary: Vec::new(),
-				fork_choice: ForkChoiceStrategy::Custom(true),
+				auxiliary: Vec::new(), // block-weight is written in block import.
+				// TODO: block-import handles fork choice and this shouldn't even have the
+				// option to specify one.
+				// https://github.com/paritytech/substrate/issues/3623
+				fork_choice: ForkChoiceStrategy::LongestChain,
```

Comment:
~~When we are authoring (before import), i.e. by this exact point, do we guarantee that nothing else gets imported in the meantime? I'm worried that we can author a primary block and won't reorg to it because in the meantime we might have imported a secondary block (with longest chain we would keep the secondary at the same height). The previous code was also susceptible to this kind of race condition if it exists.~~

In the meantime I read the code below and figured out that this logic was moved to `BabeBlockImport` which also takes care of the race condition issue.